src/odemis/util/spectrum.py

```diff
@@ -70,6 +70,10 @@ def get_wavelength_per_pixel(da):
     # MD_WL_LIST has priority
     try:
         wl = da.metadata[model.MD_WL_LIST]
+        if wl is None or len(wl) == 0:
```

Comment:
I'm a bit confused as to why this would even be None or empty, could you explain? (I don't completely understand the commit message)

Reply:
The MD_WL_LIST is updated when the spectrograph grating moves or changes. When the grating is set to a "standard" first order diffraction, then this metadata contains the wavelength for each (horizontal) pixel of the CCD. 
However, the grating can be either moved to the 0th order (aka 0mm), or even changed to a mirror grating. In this case, ideally we would remove the MD_WL_LIST metadata. However, it's not possible with `.updateMetadata()` to remove it. So instead it is set to `[]`.